frontend/src/components/DropDownGroup.jsx

```diff
@@ -0,0 +1,205 @@
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Kaikki tunnit"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckCumulativeHours = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Kaikki tunnit kumulatiivinen"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckSalary = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Palkka"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+  const userCheckBillability = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedPalkka = parsedClaims["Laskutettavat tunnit"];
+        return parsedPalkka;
+      } catch (error) {
+        return null;
+      }
+    }
+  };
+
+  const userCheckAdmin = () => {
+    if (!user) {
+      return null;
+    } else {
+      try {
+        const parsedClaims = JSON.parse(
+          user.proactiveRefresh.user.reloadUserInfo.customAttributes
+        );
+        const parsedAdmin = parsedClaims["admin"];
+        return parsedAdmin;
+      } catch (error) {
+        return null;
+      }
+    }
+    //
+  };
+
+  const items = [];
+  const itemsConstructor = () => {
+    if (userCheckHours() || userCheckCumulativeHours() || userCheckSalary()) {
+      items.push({
+        label: "Henkilöstö",
+        key: "SubMenu1",
+        children: [],
+      });
+    } else {
+      items.push({});
+    }
+    if (userCheckHours()) {
+      items[0].children.push({
+        label: <NavLink to="/monthly-working-hours">Kaikki tunnit</NavLink>,
+        key: "setting:1",
+      });
+      if (userCheckCumulativeHours()) {
+        items[0].children.push({
+          label: (
+            <NavLink to="/cumulative-monthly-working-hours">
+              Kaikki tunnit kumulatiivinen{" "}
+            </NavLink>
+          ),
+          key: "setting:2",
+        });
+      }
+      if (userCheckSalary()) {
+        items[0].children.push({
+          label: <NavLink to="/salary">Palkkaraportti</NavLink>,
+          key: "setting:3",
+        });
+      }
+    }
+    if (userCheckBillability()) {
+      items.push({
+        label: "Tilastot",
+        key: "SubMenu2",
+        children: [],
+      });
+      items[1].children.push({
```

Comment:
No need for index access here, you can add the link object directly to the children array above.